pandas/_libs/tslibs/timezones.pyx

```diff
@@ -321,6 +337,10 @@ cpdef bint tz_compare(object start, object end):
 
     """
     # GH 18523
+    if is_fixed_offset(start) and is_fixed_offset(end):
+        start_seconds = get_fixed_offset_total_seconds(start)
+        end_seconds = get_fixed_offset_total_seconds(end)
+        return start_seconds == end_seconds
```

Comment:
@mroeschke @jreback are we in agreement that two FixedOffsets of matching length should be considered equal?

Reply:
No:

```
>>> off1 = pytz.FixedOffset(420)
>>> off2 = dateutil.tz.tzoffset(None, 420*60)
>>> off1 == off2
False
```

Reply:
ok that makes sense

Reply:
Sounds reasonable.

If a user passes both a `pytz.FixedOffset` and a `dateutil.tz.tzoffset` will be coercing to one of the tzinfos? Once subtle point is if the `dateutil.tz.tzoffset` has a `name` but has the same offset as the `pytz.FixedOffset`, we should opt to keeping the dateutil instance so we don't drop the `name`.